runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -736,34 +734,35 @@ private void applyForEachAddress(BiConsumer<Long, ILogData> logDataProcessor) {
         summonNecromancer();
         nextRead = logHead;
         while (nextRead <= logTail) {
-            final long start = nextRead;
-            final long stopNotIncluded = Math.min(start + batchReadSize, logTail + 1);
-            nextRead = stopNotIncluded;
-            final Map<Long, ILogData> range = getLogData(runtime, start, stopNotIncluded);
-
-            // Sanity
-            boolean canProcessRange = true;
-            for(Map.Entry<Long, ILogData> entry : range.entrySet()) {
-                long address = entry.getKey();
-                ILogData logData = entry.getValue();
-                if (address != addressProcessed + 1) {
-                    throw new IllegalStateException("We missed an entry. It can lead to correctness issues.");
-                }
-                addressProcessed++;
+            try {
+                final long start = nextRead;
+                final long stopNotIncluded = Math.min(start + batchReadSize, logTail + 1);
+                nextRead = stopNotIncluded;
+                final Map<Long, ILogData> range = getLogData(runtime, start, stopNotIncluded);
+
+                // Sanity
+                for (Map.Entry<Long, ILogData> entry : range.entrySet()) {
+                    long address = entry.getKey();
+                    ILogData logData = entry.getValue();
+                    if (address != addressProcessed + 1) {
+                        throw new IllegalStateException("We missed an entry. It can lead to correctness issues.");
+                    }
+                    addressProcessed++;
 
-                if (logData.getType() == DataType.TRIMMED) {
-                    log.warn("applyForEachAddress[{}, start={}] address is trimmed", address, logHead);
-                    handleRetry();
-                    canProcessRange = false;
-                    break;
-                }
+                    if (logData.getType() == DataType.TRIMMED) {
```

Comment:
Will this ever happen? Trimmed addresses are detected by the AddressSpace layer through getLogData.

Reply:
You are right, it shouldn't happen, that's why we throw IllegalStateException. Since it was in prev version, decided not to get rid of this part